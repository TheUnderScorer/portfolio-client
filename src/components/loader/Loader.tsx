import * as React from 'react';
import { useEffect, useState } from 'react';
import { connect, useSelector } from 'react-redux';
import BaseStore from '../../types/stores/BaseStore';
import { getPrimary } from '../styled/colors';
import LoaderProps from './types/LoaderProps';
import styled from 'styled-components';

const LoaderSvg = styled.svg<LoaderProps>`
    transition: opacity .3s;
    opacity: 0;
    visibility: hidden;
    height: ${ props => props.height };
    width: ${ props => props.width };
    
    ${ props => props.active && `
        opacity: 1;
        visibility: visible;
    ` }
`;

const Loader = ( { active = false, height = '200px', width = '200px' }: LoaderProps ) => {

    const themeMode = useSelector( ( store: BaseStore ) => store.theme.mode );
    const [ color, setColor ] = useState( getPrimary( themeMode ) );

    useEffect( () => {
        setColor(
            getPrimary( themeMode )
        )
    }, [ themeMode ] );

    return (
        <LoaderSvg width={ width } height={ height } active={ active } xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100" preserveAspectRatio="xMidYMid" className="lds-eclipse" style={ { background: 'none' } }>
            <path stroke="none" d="M10 50A40 40 0 0 0 90 50A40 42 0 0 1 10 50" fill={ color } transform="rotate(281.917 50 51)">
                <animateTransform attributeName="transform" type="rotate" calcMode="linear" values="0 50 51;360 50 51" keyTimes="0;1" dur="1s" begin="0s" repeatCount="indefinite"/>
            </path>
        </LoaderSvg>
    )
};

export default connect()( Loader );
