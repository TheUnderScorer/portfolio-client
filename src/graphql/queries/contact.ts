import { gql } from 'apollo-boost';

export const SEND = gql`
    mutation Send($input: ContactInput!) {
        send(contactInput: $input) {
            id
        }
    }
`;
