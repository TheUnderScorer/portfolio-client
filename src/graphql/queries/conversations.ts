import { CONVERSATION_FRAGMENT, MESSAGE_FRAGMENT } from '../fragments/conversations';
import gql from 'graphql-tag';

export const MY_CONVERSATIONS = gql`
    query GetConversations($page: Int = 1, $perPage: Int = 15) {
        conversations:myConversations(page: $page, perPage:$perPage) {
            id,
            title,
            messages {
                id,
                content,
                author {
                    id,
                    name,
                },
                createdAt
            }
        }
    }
`;

export const MY_CONVERSATION = gql`
    query MyConversation($page: Int = 1, $perPage: Int = 30) {
        conversation:getCurrentConversation{
            ...ConversationFragment
            messages(perPage: $perPage, page: $page) {
                ...MessageFragment
            }
        }
    }
    ${MESSAGE_FRAGMENT}
    ${CONVERSATION_FRAGMENT}
`;


