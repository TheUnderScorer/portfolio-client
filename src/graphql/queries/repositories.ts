import { gql } from 'apollo-boost';

export const GET_REPOSITORIES = gql`
    query GetRepositories($first: Int)
    {
        viewer {
            repositories(first: $first, orderBy: {direction: DESC, field: CREATED_AT}) {
                nodes {
                    name,
                    createdAt,
                    description,
                    url
                }
            }
        }
    }
`;
