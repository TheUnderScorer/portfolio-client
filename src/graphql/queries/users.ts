import { gql } from 'apollo-boost';

export const GET_ME = gql`
    {
        user:me {
            id,
            name,
            email,
            token {
                value
            }
        }
    }
`;

export const CREATE_USER = gql`
    mutation CreateUser ($input: UserInput){
        user:createUser(userInput: $input) {
            id,
            name,
            email,
            token {
                value
            }
        }
    }
`;

export const UPDATE_USER = gql`
    mutation UpdateUser ($input: UserInput!) {
        updateUser(userInput: $input) {
            id,
            name,
            email
        }
    }
`;

export const UPDATE_LOGIN_DATE = gql`
    mutation {
        user:updateLoginDate {
            lastLogin
        }
    }
`;

export const UPDATE_ME = gql`
    mutation UpdateMe ($input: UserInput!) {
        updateMe(userInput: $input) {
            id,
            name,
            email
        }
    }
`;
