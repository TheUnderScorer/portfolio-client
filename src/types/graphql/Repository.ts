export default interface Repository
{
    createdAt: string;
    description: string;
    name: string;
    url: string;
}
