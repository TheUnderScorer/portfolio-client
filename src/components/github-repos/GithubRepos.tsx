import * as React from 'react';
import { Fragment } from 'react';
import { useQuery } from '@apollo/react-hooks';
import { GET_REPOSITORIES } from '../../graphql/queries/repositories';
import { GetRepositoriesResult } from '../../types/graphql/Queries';
import Loader from '../loader/Loader';
import { Divider, Grid, List } from '@material-ui/core';
import IconMessage from '../icon-message/IconMessage';
import { FaIcon, Text } from '../styled/typography';
import { faFrown } from '@fortawesome/free-regular-svg-icons';
import { isEmpty } from 'lodash';
import GithubReposPops from './types/GithubReposPops';
import { Title } from './styled';
import GithubRepo from './GithubRepo';


const GithubRepos = ( { queryVariables: { first = 5 } }: GithubReposPops ) =>
{
    const { data, loading, error } = useQuery<GetRepositoriesResult>( GET_REPOSITORIES, {
        variables: {
            first,
        },
    } );

    return (
        <Grid container>
            <Title>
                My latest repositories
            </Title>
            { loading &&
              <Grid item xs={ 12 }>
                  <Loader svgProps={ { width: '50px', height: '40px' } } active={ loading }/>
              </Grid>
            }
            { error &&
              <Grid item xs={ 12 }>
                  <IconMessage icon={
                      <FaIcon icon={ faFrown }/> } title="Request error!">
                      <Text>
                          { error.message }
                      </Text>
                  </IconMessage>
              </Grid>
            }
            { data && !isEmpty( data ) &&
              <Grid item xs={ 12 }>
                  <List>
                      { data.viewer.repositories.nodes.map( ( repo, index ) =>
                          <Fragment key={ index }>
                              <GithubRepo { ...repo } />
                              { index < data.viewer.repositories.nodes.length - 1 &&
                                <Divider component="li" variant="inset"/> }
                          </Fragment>
                      ) }
                  </List>
              </Grid>
            }
        </Grid>
    )

};

export default GithubRepos;
