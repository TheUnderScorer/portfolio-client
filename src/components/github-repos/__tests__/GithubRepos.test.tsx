import { MockedResponse, wait } from '@apollo/react-testing';
import { GET_REPOSITORIES } from '../../../graphql/queries/repositories';
import Repository from '../../../types/graphql/Repository';
import moment from 'moment';
import { mountWithStoreAndApollo } from '../../../tests/renderer';
import GithubRepos from '../GithubRepos';
import React from 'react';
import Loader from '../../loader/Loader';
import { act } from 'react-dom/test-utils';

describe( 'GithubRepos', () =>
{
    const repositories: Repository[] = [
        {
            name:        'Test repo 1',
            description: 'Just a test repo',
            url:         'http://localhost/test-repo',
            createdAt:   moment().toISOString()
        }
    ];

    const mocks: MockedResponse[] = [
        {
            request: {
                query:     GET_REPOSITORIES,
                variables: {
                    first: 5
                },
            },
            delay:   200,
            result:  {
                data: {
                    viewer: {
                        repositories: {
                            nodes: repositories
                        }
                    }
                }
            }
        }
    ];

    const renderComponent = () =>
        mountWithStoreAndApollo(
            <GithubRepos queryVariables={ { first: 5 } }/>,
            {
                theme: {
                    mode: 'white'
                }
            },
            {
                mocks,
                addTypename: true
            }
        );

    it( 'Renders without crashing', () =>
    {
        renderComponent();
    } );

    it( 'Shows loader when query is loading', async () =>
    {
        const { component } = renderComponent();

        await wait( 100 );

        const loader = component.find( Loader );
        const loaderProps = loader.props();

        expect( loader ).toHaveLength( 1 );
        expect( loaderProps.active ).toBeTruthy();
    } );

    it( 'Hides loader when query has loaded', async () =>
    {
        const { component } = renderComponent();

        await act( async () =>
        {
            await wait( 1000 );
        } );

        const loader = component.update().find( Loader );

        expect( loader ).toEqual( {} );
    } )
} );
