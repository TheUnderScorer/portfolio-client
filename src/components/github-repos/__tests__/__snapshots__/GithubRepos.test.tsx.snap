// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubRepos Hides loader when query has loaded 1`] = `"<div class=\\"MuiGrid-root MuiGrid-container\\"><h4 class=\\"sc-bZQynM sc-dnqmqq sc-gqjmRU sc-jKJlTe jfJzG\\">My latest repositories</h4><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><ul class=\\"MuiList-root MuiList-padding\\"><li class=\\"MuiListItem-container\\"><a class=\\"MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart MuiListItem-secondaryAction\\" target=\\"__blank\\" href=\\"http://localhost/test-repo\\"><div class=\\"MuiListItemIcon-root MuiListItemIcon-alignItemsFlexStart\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fab\\" data-icon=\\"github\\" class=\\"svg-inline--fa fa-github fa-w-16 sc-cSHVUG sc-kAzzGY lfldqO\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 496 512\\"><path fill=\\"currentColor\\" d=\\"M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z\\"></path></svg></div><div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\"><span class=\\"sc-htpNat gNGRIp\\">Test repo 1</span></span><p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary\\"><span class=\\"sc-htpNat hrkesl\\">Just a test repo</span><small class=\\"sc-EHOje jrGvfy\\"> - 2019-09-14 19:30:04</small></p></div></a><div class=\\"MuiListItemSecondaryAction-root\\"><button mode=\\"secondary\\" class=\\"sc-eNQAEJ sc-kEYyzF gUXTZx\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"external-link-alt\\" class=\\"svg-inline--fa fa-external-link-alt fa-w-18 sc-cSHVUG sc-kAzzGY lfldqO\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\"><path fill=\\"currentColor\\" d=\\"M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z\\"></path></svg></button></div></li></ul></div></div>"`;

exports[`GithubRepos Renders without crashing 1`] = `"<div class=\\"MuiGrid-root MuiGrid-container\\"><h4 class=\\"sc-bZQynM sc-dnqmqq sc-gqjmRU sc-jKJlTe jfJzG\\">My latest repositories</h4><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"sc-bdVaJa ewRZkf\\" width=\\"100%\\" height=\\"100%\\"><svg width=\\"50px\\" height=\\"40px\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 100 100\\" preserveAspectRatio=\\"xMidYMid\\" class=\\"lds-eclipse sc-bwzfXH gJdAKd\\" style=\\"background: none;\\"><path stroke=\\"none\\" d=\\"M10 50A40 40 0 0 0 90 50A40 42 0 0 1 10 50\\" fill=\\"#FD5A36\\" transform=\\"rotate(281.917 50 51)\\"><animateTransform attributeName=\\"transform\\" type=\\"rotate\\" calcMode=\\"linear\\" values=\\"0 50 51;360 50 51\\" keyTimes=\\"0;1\\" dur=\\"1s\\" begin=\\"0s\\" repeatCount=\\"indefinite\\"></animateTransform></path></svg></div></div></div>"`;
